tests/TestCase/ORM/MarshallerTest.php

```diff
@@ -1374,6 +1374,32 @@ public function testMergeUnchangedNullValue()
         $this->assertFalse($entity->isDirty('body'), 'unchanged null should not be dirty');
     }
 
+    /**
+     * Test merge() doesn't dirty objects which are equal.
+     *
+     * @return void
+     */
+    public function testMergeWithSameObjectValue()
+    {
+        $created = new FrozenTime('2020-10-29');
+        $entity = new Entity([
+            'comment' => 'foo',
+            'created' => $created,
+        ]);
+        $entity->setAccess('*', true);
+        $entity->setNew(false);
+        $entity->clean();
+
+        $data = [
+            'comment' => 'bar',
+            'created' => clone $created,
```

Comment:
Should we create a different type here?

Reply:
Why? I think this captures the regression well enough.

Reply:
I missed that cloning alone would fail the `===` test. This should work.